src/pkg/services/m365/custom/drive_item.go

```diff
@@ -0,0 +1,339 @@
+// Disable revive linter since any structs in this file will expose the same
+// funcs as the original structs in the msgraph-sdk-go package, which do not
+// follow some of the golint rules.
+//
+//nolint:revive
+package custom
+
+import (
+	"strings"
+	"time"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/common/str"
+)
+
+// Replica of models.DriveItemable
+type LiteDriveItemable interface {
```

Comment:
Nit: to help clarify the relationship between this struct and `models.DriveItemable`.  Since the "lightness" largely comes from the transformation to a local struct, and not a reduced schema.
```suggestion
type ModelDriveItemable interface {
```

Reply:
True, but I didn't want to use `custom.DriveItemable` name since it can be easily confused with `models.DriveItemable`. What do you think? Is it a fair concern?

Reply:
Personal preference: if we don't intend to have this item comply with the complete `models.DriveItemable` interface (as you mentioned [here](https://github.com/alcionai/corso/pull/4783#discussion_r1414296674)), then we don't need this interface at all.  Seems like we're fine to transform the graph object into a concrete struct (`custom.DriveItem`) and hand that around directly.

Reply:
Yeah, that's a reasonable thing to do. Thanks! Let me make this change. I think Vaibhav also had a similar comment.